xapian-core/backends/glass/glass_version.cc

```diff
@@ -113,7 +113,7 @@ GlassVersion::read()
 	if (rare(fd_in < 0)) {
 	    string msg = filename;
 	    msg += ": Failed to open glass revision file for reading";
-	    throw Xapian::DatabaseOpeningError(msg, errno);
+	    throw Xapian::DatabaseNotFoundError(msg, errno);
```

Comment:
This case on the other hand does seem like "not found" - e.g. you can get it if you try to open a directory which isn't a Xapian DB and specify `Xapian::DB_BACKEND_GLASS` in the flags.

Reply:
> This case on the other hand does seem like "not found" - e.g. you can get it if you try to open a directory which isn't a Xapian DB and specify `Xapian::DB_BACKEND_GLASS` in the flags.
>
@ojwb the value of fd_in could be negative for several reasons other than "not found", so shouldn't it be `DatabaseOpeningError`?

Reply:
Probably "not found" is the most common reason, but there are other possibilities as you say - not having the necessary permissions is also a common one.

Checking `errno` and picking which of the two to throw would be best I guess - `ENOENT` is the obvious value to check, but `ENOTDIR` should probably also give `DatabaseNotFoundError` (you'd get that if a parent directory in the path is actual a file, e.g. trying to open `/home/olly/foo/bar.db` when `/home/olly/foo` is a file, which is very similar to the case of trying to open `/home/olly/foo/bar.db` when `/home/olly/foo` doesn't exist).